hoodie-client/src/main/java/com/uber/hoodie/HoodieWriteClient.java

```diff
@@ -122,6 +123,20 @@ public HoodieWriteClient(JavaSparkContext jsc, HoodieWriteConfig clientConfig,
     }
   }
 
+  @VisibleForTesting
```

Comment:
The existing HoodieWriteClient constructor may use the new constructor to save duplicate code.
```
public HoodieWriteClient(JavaSparkContext jsc, HoodieWriteConfig clientConfig,
       boolean rollbackInFlight) {	       boolean rollbackInFlight) {
```

Reply:
Good point.. @kaushikd49 can we fix that as well.. 

Defining a new package private constructor is much cleaner.. Nicely done!

Reply:
@vinothchandar @jianxu Existing public constructor does not take index as arg (which is a good thing since production code should not pass index), so cannot re-use the existing constructor. Also cannot refactor common code to a init() method because the instance variables are all finals. What do you guys think?

Reply:
Nvm got it :)